nebula_ros/src/hesai/hw_monitor_wrapper.cpp

```diff
@@ -53,28 +51,19 @@ HesaiHwMonitorWrapper::HesaiHwMonitorWrapper(
                          config->sensor_model != drivers::SensorModel::HESAI_PANDAR40P &&
                          config->sensor_model != drivers::SensorModel::HESAI_PANDAR64;
 
-  std::shared_ptr<HesaiInventoryBase> inventory = hw_interface->get_inventory();
-  RCLCPP_INFO_STREAM(logger_, "Inventory info: " << *inventory);
-  json inventory_json = inventory->to_json();
-
-  std::string model = inventory_json.at("model");
-  std::string serial = inventory_json.at("sn");
-  auto hardware_id = model + ": " + serial;
-  diagnostics_updater_.setHardwareID(hardware_id);
-  RCLCPP_INFO_STREAM(logger_, "Hardware ID: " + hardware_id);
-
-  initialize_hesai_diagnostics(monitor_enabled);
+  initialize_hesai_diagnostics(diagnostic_updater, monitor_enabled);
 }
 
-void HesaiHwMonitorWrapper::initialize_hesai_diagnostics(bool monitor_enabled)
+void HesaiHwMonitorWrapper::initialize_hesai_diagnostics(
+  diagnostic_updater::Updater & diagnostic_updater, bool monitor_enabled)
 {
   using std::chrono_literals::operator""s;
   std::ostringstream os;
-  diagnostics_updater_.add("hesai_status", this, &HesaiHwMonitorWrapper::hesai_check_status);
-  diagnostics_updater_.add("hesai_ptp", this, &HesaiHwMonitorWrapper::hesai_check_ptp);
-  diagnostics_updater_.add(
+  diagnostic_updater.add("hesai_status", this, &HesaiHwMonitorWrapper::hesai_check_status);
```

Comment:
Since our design now has human-readable task names, it would be good to align for these items.

Reply:
We should probably also combine them into a single `DiagnosticStatus`, rather than having four separate ones.
The key-value pairs should also receive updated names, so I'd suggest we do this in a separate PR?

Reply:
Agreed :+1: